android/java/org/chromium/chrome/browser/rewards/BraveRewardsCreatorPanelFragment.java

```diff
@@ -0,0 +1,356 @@
+            @Override
+            public void onClick(View v) {
+                String url = (String) v.getTag();
+                openLink(url);
+            }
+        });
+    }
+
+    private void openLink(String url) {
+        Intent intent = new Intent();
+        intent.putExtra(BraveActivity.OPEN_URL, url);
+        getActivity().setResult(Activity.RESULT_OK, intent);
+        getActivity().finish();
+    }
+
+    private void showVerifiedIcon(View view) {
+        @PublisherStatus
+        int pubStatus = mBraveRewardsNativeWorker.GetPublisherStatus(currentTabId_);
+
+        if (pubStatus == BraveRewardsPublisher.CONNECTED
+                || pubStatus == BraveRewardsPublisher.UPHOLD_VERIFIED
+                || pubStatus == BraveRewardsPublisher.BITFLYER_VERIFIED
+                || pubStatus == BraveRewardsPublisher.GEMINI_VERIFIED) {
+            view.findViewById(R.id.publisher_favicon_verified).setVisibility(View.VISIBLE);
+        }
+    }
+
+    private void retrivePublisherName(View view) {
+        String publisherName = mBraveRewardsNativeWorker.GetPublisherName(currentTabId_);
+        TextView publisher = (TextView) view.findViewById(R.id.publisher_name);
+        publisher.setText(publisherName);
+    }
+
+    private void retriveFavIcon() {
+        String publisherFavIconURL =
+                mBraveRewardsNativeWorker.GetPublisherFavIconURL(currentTabId_);
+        Tab currentActiveTab = BraveRewardsHelper.currentActiveChromeTabbedActivityTab();
+        String url = currentActiveTab.getUrl().getSpec();
+        String favicon_url = (publisherFavIconURL.isEmpty()) ? url : publisherFavIconURL;
+        mIconFetcher = new BraveRewardsHelper(currentActiveTab);
+        mIconFetcher.retrieveLargeIcon(favicon_url, this);
+    }
+
+    private void setDescription(View view) {
+        if (mBannerInfo == null) return;
+        String description = mBannerInfo.getDescription();
+        if (description != null && !description.isEmpty()) {
+            TextView descriptionTextView =
+                    (TextView) view.findViewById(R.id.rewards_banner_description);
+            descriptionTextView.setText(description);
+        }
+    }
+
+    private void setTitle(View view) {
+        if (mBannerInfo == null) return;
+        String title = mBannerInfo.getTitle();
+        if (title != null && !title.isEmpty()) {
+            TextView titleTextView = (TextView) view.findViewById(R.id.banner_title);
+            titleTextView.setText(title);
+        }
+    }
+
+    private void SetFavIcon(Bitmap bmp) {
+        if (bmp != null) {
+            getActivity().runOnUiThread(new Runnable() {
+                @Override
+                public void run() {
+                    ImageView iv = (ImageView) getActivity().findViewById(R.id.publisher_favicon);
+                    int nPx = BraveRewardsHelper.dp2px(PUBLISHER_ICON_SIDE_LEN);
+                    Bitmap resized = Bitmap.createScaledBitmap(bmp, nPx, nPx, true);
```

Comment:
shouldn't that be call outside of the UI thread?

Reply:
it's called inside runOnUiThread

Reply:
moved this line to current thread.